### ISSUE TITLE
Update examples to use styles and Font Awesome

### ISSUE THREAD

**Issue by agarcialeon** [joinfaces/joinfaces#734]
Hi everyone,

I've been using JoinFaces lately and after some problems following tutorials trying to setup a simple project, now I have finished integrating my first web app with JoinFaces.

However now I'm dealing with styling of components and trying to integrate font awesome icons, but no luck for the moment with the latter.

What I'm asking for is if it's possible to update the example projects to use font awesome and styling instead of just using simple components because I find the documentation a little empty about this.

**Comment by melloware** [comment]
@agarcialeon Which Faces UI library do you use?  Primefaces?

**Comment by agarcialeon** [comment]
Yes, Primefaces. I have a basic project with spring-boot-starter-web, spring-boot-starter-test and primefaces-spring-boot-starter.

For the moment, I'm using JAR packaging but in the future I'll need a WAR with webapp folder as in other projects from my company.

**Comment by melloware** [comment]
OK I use this too and I simply add this to my main layout template:

```xml
<h:outputStylesheet library="webjars" name="font-awesome/5.10.1/css/all.min-jsf.css" />
<h:outputStylesheet library="webjars" name="font-awesome/5.10.1/css/v4-shims.min-jsf.css" />
```

And in my pom.xml

```xml
<dependency>
     <groupId>org.webjars</groupId>
     <artifactId>font-awesome</artifactId>
     <version>5.10.1</version>
</dependency>
```

That fully enables the latest FontAwesome 5 into your application.

**Comment by agarcialeon** [comment]
Ok, but then Font Awesome should be added as a dependency? 
I thought it was already referenced by the dependency of JoinFaces.
Apart from that, if I want to include PrimeFlex, is the process the same as you stated?

**Comment by melloware** [comment]
OK so two issues.  

1. PrimeFaces includes FontAwesome 4.7 inside its JAR which you can enable in JoinFaces in your application.yaml...

```yaml
joinfaces:
  jsf:
    project-stage: Development
  primefaces:
    font-awesome: true
```
However PF only includes the old FontAwesome 4.7 with about 500 icons.  My solution above allows you to include FontAwesome 5 which includes about 8000 icons.  Up to you.

2. PrimeFlex 
- you need to download primeflex.css from here: https://github.com/primefaces/primeflex
- Add primeflex.css to your project folder `\src\main\resources\META-INF\resources\css`
- include it on your page `<h:outputStylesheet name="css/primeflex.css"/>`

**Comment by larsgrefer** [comment]
@agarcialeon Have the hints from @melloware helped you to solve your problems or do you still have questions?

**Comment by agarcialeon** [comment]
Yes, it works! Sorry for not commenting about it. You can close the issue. And thanks again.
